### PR TITLE
fix(parser): Unable to parse json in one line

### PR DESCRIPTION
**Pull request by iamxiaojianzheng**
about this https://github.com/oysandvik94/curl.nvim/issues/41

### CODE REVIEW

File: lua/curl/parser.lua [hunk 1]
```diff
@@ -68,9 +68,13 @@ local format_command_for_curl = function(lines)
 		if found_first_json_char(json_nesting_stack, cleaned_line) then
 			opening_json_char = cleaned_line:sub(1, 1)
 			closing_json_char = opening_json_char == "[" and "]" or "}"
-			is_line_json_close(json_nesting_stack, opening_json_char, closing_json_char, cleaned_line)
-
 			cleaned_line = "'" .. cleaned_line
+
+			is_line_json_close(json_nesting_stack, opening_json_char, closing_json_char, cleaned_line)
```
Comment (oysandvik94) [inline lua/curl/parser.lua:73]: This function returns a boolean whether the line contains a json closer, however the result is never used. 
Is the case fixed if you do the same as in the else branch instead?
```lua

**Review by oysandvik94 — CHANGES_REQUESTED** [review]
See the comment.

Also, I think we should have a test for the case that fails. 
It can be added to "parser_spec.lua", there are other examples in that file.

**Comment by iamxiaojianzheng** [comment]
After reading the code, I found that although this method did not use its return value, it operated on json_nesting_stack, which was critical.

**Comment by iamxiaojianzheng** [comment]
> See the comment.
> 
> 
> 
> Also, I think we should have a test for the case that fails. 
> 
> It can be added to "parser_spec.lua", there are other examples in that file.

OK, I'll add the test case later.

**Comment by oysandvik94** [comment]
> After reading the code, I found that although this method did not use its return value, it operated on json_nesting_stack, which was critical.

That's right, might not be the best design from my part hehe.

But this check:
#json_nesting_stack == 0
Is done inside "is_line_json_close()", and the return value of that evaluation, so I think you can assert the result of that function instead of asserting the size of the stack afterwards. I think that will end up in more readable code.

**Comment by iamxiaojianzheng** [comment]
I have finished the revision

File: lua/curl/parser.lua [hunk 2]
```diff
@@ -68,9 +68,12 @@ local format_command_for_curl = function(lines)
 		if found_first_json_char(json_nesting_stack, cleaned_line) then
 			opening_json_char = cleaned_line:sub(1, 1)
 			closing_json_char = opening_json_char == "[" and "]" or "}"
-			is_line_json_close(json_nesting_stack, opening_json_char, closing_json_char, cleaned_line)
-
 			cleaned_line = "'" .. cleaned_line
+
+			-- if the all json content is in one line
```
Comment (oysandvik94) [inline lua/curl/parser.lua:73]: Dont need this comment, this if check will work for all cases

**Review by oysandvik94 — CHANGES_REQUESTED** [review]

**Comment by oysandvik94** [comment]
Looks very good! I just had a little nitpick

**Review by oysandvik94 — APPROVED** [review]

File: lua/curl/parser.lua (hunk 2 above)
Comment (iamxiaojianzheng) [reply lua/curl/parser.lua]: That makes sense. It's been taken care of.

**Comment by oysandvik94** [comment]
Thank you so much for your contribution :smile_cat:

**Comment by iamxiaojianzheng** [comment]
> Thank you so much for your contribution 😸

Look forward to your refactoring. 😀

**Pull request merged**